swiftpkg/internal/artifact_infos.bzl

```diff
@@ -89,18 +89,39 @@ def _new_framework_info_from_files(repository_ctx, path):
     )
     if len(binary_files) == 0:
         fail("No binary files were found for framework at {}".format(path))
-    file_type = repository_files.file_type(repository_ctx, binary_files[0])
-    if file_type.find("ar archive random library") > 0:
-        link_type = link_types.static
-    elif file_type.find("dynamically linked shared library"):
-        link_type = link_types.dynamic
-    else:
-        link_type = link_types.unknown
+    link_type = _link_type(repository_ctx, binary_files[0])
+
     return _new_framework_info(
         path = path,
         link_type = link_type,
     )
 
+def _link_type(repository_ctx, path):
+    """Determine the link type for the framework binary file.
+
+    Args:
+        repository_ctx: A `repository_ctx` instance.
+        path: The path to a framework binary file under a `XXX.framework`
+            directory as a `string`.
+
+    Returns:
+        The link type for the framework as a `string`.
+    """
+    file_type = repository_files.file_type(repository_ctx, path)
+
+    # static Examples:
+    #   current ar archive random library
+    #   current ar archive
+    # dynamic Examples:
+    #   dynamically linked shared library
+    if file_type.find("ar archive") >= 0:
+        link_type = link_types.static
+    elif file_type.find("dynamic") >= 0:
+        link_type = link_types.dynamic
+    else:
+        link_type = link_types.unknown
+    return link_type
```

Comment:
nit: could just return in each branch vs. setting `link_type`

Reply:
Done.